src/en/honkaiimpact/src/eu/kanade/tachiyomi/extension/en/honkaiimpact/Honkaiimpact.kt

```diff
@@ -22,63 +25,72 @@ import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-// Info - Based of BH3
-// This is the english version of the site
 class Honkaiimpact : ParsedHttpSource() {
 
     override val name = "Honkai Impact 3rd"
-
     override val baseUrl = "https://manga.honkaiimpact3.com"
-
     override val lang = "en"
-
     override val supportsLatest = false
 
+    private var searchQuery = ""
+    private val json: Json by injectLazy()
+
     override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        // .addInterceptor(filteringInterceptor)
         .connectTimeout(1, TimeUnit.MINUTES)
         .readTimeout(1, TimeUnit.MINUTES)
         .retryOnConnectionFailure(true)
         .followRedirects(true)
         .build()
 
-    private val json: Json by injectLazy()
-
     // Popular
     override fun popularMangaSelector() = "a[href*=book]"
-
     override fun popularMangaNextPageSelector(): String? = null
-
-    override fun popularMangaRequest(page: Int) = GET("$baseUrl/book", headers)
-
+    override fun popularMangaRequest(page: Int) =
+        GET("$baseUrl/book", headers) // .newBuilder().tag(String::class.java, searchQuery).build()
     override fun popularMangaFromElement(element: Element) = mangaFromElement(element)
 
     // Latest
     override fun latestUpdatesSelector() = throw UnsupportedOperationException()
-
     override fun latestUpdatesNextPageSelector(): String? = null
-
     override fun latestUpdatesRequest(page: Int) = throw UnsupportedOperationException()
-
-    override fun latestUpdatesFromElement(element: Element) = mangaFromElement(element)
+    override fun latestUpdatesFromElement(element: Element) = throw UnsupportedOperationException()
 
     // Search
-    override fun searchMangaSelector() = throw UnsupportedOperationException()
-
+    override fun searchMangaSelector() = "a[href*=book]"
     override fun searchMangaNextPageSelector(): String? = null
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        searchQuery = query
```

Comment:
This is arguably worse. Prone to race conditions. I did not consider the full implications of my previous comment. Use the lower level function - [`fetchSearchManga()`](https://github.com/mihonapp/mihon/blob/e8c6e3364dded0cb21df572be175cbb01adc243f/source-api/src/commonMain/kotlin/eu/kanade/tachiyomi/source/online/HttpSource.kt#L134-L160) - here you will have the query in scope when parsing the response.

Reply:
or just pass the query in url fragment or request Tag